### PR TITLE
Update defining_steps.md

### PR DESCRIPTION
**Pull request by stephanieatte**
Users keep getting confused about Build States and Job states. It makes sense to give description of jobs state in our docs

### CODE REVIEW

File: pages/pipelines/defining_steps.md [hunk 1]
```diff
@@ -124,6 +123,31 @@ When you run a pipeline, a build is created. Each of the steps in the pipeline e
 
 <%= image "job-states.svg", size: "1482x690", alt: "Job state diagram" %>
 
+Job State             | Description
```
Comment (mbelton-buildkite) [inline pages/pipelines/defining_steps.md:126]: Style: Use sentence casing.

```suggestion
Job state             | Description
```

File: pages/pipelines/defining_steps.md [hunk 2]
```diff
@@ -124,6 +123,31 @@ When you run a pipeline, a build is created. Each of the steps in the pipeline e
...
+----------------------| -----------------------------------------
+PENDING               | The job has just been created and doesn't have a state yet
```
Comment (mbelton-buildkite) [inline pages/pipelines/defining_steps.md:128]: Style:

- In the diagram above and text below it looks like these are normally written in lower case and using code styling. I think this should be consistent with that unless there's a good reason not to.
- End each description with a period.

```suggestion
`pending`               | The job has just been created and doesn't have a state yet.
```

^ This applies to each row.

**Review by mbelton-buildkite — COMMENTED** [review]
This is a good addition! Just a few style nitpicks to fix.

**Review by gilesgas — APPROVED** [review]
Thanks very much for your contribution @stephanieatte !

This is now ready to merge 🎉

**Pull request merged**